tests/functional/build/test_build.py

```diff
@@ -0,0 +1,192 @@
+import pytest
+
+from dbt.tests.util import run_dbt, copy_file
+
+
+from tests.functional.build.fixtures import (  # noqa: F401
+    snapshots,
+    tests_failing,
+    models,
+    models_failing,
+    seeds,
+    models_circular_relationship,
+    models_simple_blocking,
+    test_files,
+    models_interdependent,
+    project_files,
+)
+
+
+# from test.integration.base import DBTIntegrationTest, use_profile, normalize
+from test.integration.base import normalize
+import yaml
+import os
+
+
+class TestBuildBase:
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "config-version": 2,
+            "snapshot-paths": ["snapshots"],
+            "seed-paths": ["seeds"],
+            "seeds": {
+                "quote_columns": False,
+            },
+        }
+
+    def build(self, expect_pass=True, extra_args=None, **kwargs):
+        args = ["build"]
+        if kwargs:
+            args.extend(("--args", yaml.safe_dump(kwargs)))
+        if extra_args:
+            args.extend(extra_args)
+
+        return run_dbt(args, expect_pass=expect_pass)
+
+
+class TestPassingBuild(TestBuildBase):
+    @pytest.fixture(scope="class")
+    def model_path(self):
+        return "models"
+
+    def test__build_happy_path(
+        self,
+        project,
+    ):
+        self.build(expect_pass=True)
+
+
+class TestFailingBuild(TestBuildBase):
+    @pytest.fixture(scope="class")
+    def models(self, models_failing):  # noqa: F811
+        return {
+            "model_3.sql": models_failing["model_3.sql"],
+            "model_2.sql": models_failing["model_2.sql"],
+            "test.yml": models_failing["test.yml"],
+            "model_0.sql": models_failing["model_0.sql"],
+            "model_1.sql": models_failing["model_1.sql"],
+            "model_99.sql": models_failing["model_99.sql"],
+        }
+
+    def test__build_sad_path(
```

Comment:
All the test names should be single underscore ie `test_build_sad_path` to be consistent with all other tests.

Reply:
Sure, another conversion script thing here @ChenyuLInx